products/privateca/api.yaml

```diff
@@ -0,0 +1,265 @@
+            - !ruby/object:Api::Type::String
+              name: 'organization'
+              description: The organization of the subject.
+            - !ruby/object:Api::Type::String
+              name: 'organizationalUnit'
+              description: The organizational unit of the subject.
+            - !ruby/object:Api::Type::String
+              name: 'locality'
+              description: The locality or city of the subject.
+            - !ruby/object:Api::Type::String
+              name: 'province'
+              description: The province, territory, or regional state of the subject.
+            - !ruby/object:Api::Type::String
+              name: 'streetAddress'
+              description: The street address of the subject.
+            - !ruby/object:Api::Type::String
+              name: 'postalCode'
+              description: The postal code of the subject.
+          - !ruby/object:Api::Type::String
+            name: 'commonName'
+            description: The common name of the distinguished name.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'subjectAltName'
+            description: The subject alternative name fields.
+            properties:
+            - !ruby/object:Api::Type::Array
+              name: 'dnsNames'
+              description: Contains only valid, fully-qualified host names.
+              item_type: Api::Type::String
+            - !ruby/object:Api::Type::Array
+              name: 'uris'
+              description: Contains only valid RFC 3986 URIs.
+              item_type: Api::Type::String
+            - !ruby/object:Api::Type::Array
+              name: 'emailAddresses'
+              description: Contains only valid RFC 2822 E-mail addresses.
+              item_type: Api::Type::String
+            - !ruby/object:Api::Type::Array
+              name: 'ipAddresses'
+              description: Contains only valid 32-bit IPv4 addresses or RFC 4291 IPv6 addresses.
+              item_type: Api::Type::String
+        - !ruby/object:Api::Type::NestedObject
+          name: 'reusableConfig'
+          description: |
+              Specifies some of the values in a certificate that are related to the subject.
+          required: true
+          properties:
+            - !ruby/object:Api::Type::String
+              name: 'reusableConfig'
+              description: |
+                A resource path to a ReusableConfig in the format
+                projects/*/locations/*/reusableConfigs/*.
+      - !ruby/object:Api::Type::String
+        name: 'lifetime'
+        description: |
+          The desired lifetime of the CA certificate. Used to create the "notBeforeTime" and
+          "notAfterTime" fields inside an X.509 certificate. A duration in seconds with up to nine
+          fractional digits, terminated by 's'. Example: "3.5s".
+        default_value: 315360000s  # 10 years
+      - !ruby/object:Api::Type::NestedObject
+        name: 'keySpec'
```

Comment:
missing property [cloudKmsKeyVersion](https://cloud.google.com/certificate-authority-service/docs/reference/rest/v1beta1/projects.locations.certificateAuthorities#KeyVersionSpec) .... docs say this is required..

Reply:
I think this is another instance of the oneof marking both as required when only one should be specified (and I've ommitted the more advanced/complex config in this initial version).

Reply:
gotcha gotcha